### PR TITLE
Search bar fixes

### PR DESCRIPTION
**Pull request by bhollis**
A bevy of search bar fixes, all at once. Separated into individual commits for easy browsing. The highlights:

* Disable `Home` and `End` key behavior, returning them to the default.
* Make the save-search button an enticing DIM orange.
* Remove attempts to immediately apply the debounced query - this was the "input jank" @nev-r was seeing.
* Build a new debugging hook called `useWhatChanged` that can tell you why a component is re-rendering.
* Use that component and some _forbidden techniques_ to get down to one render per keystroke.
* Go back to the recommended way of handling state in Downshift, which made things a lot simpler.
* Extract the special search bar actions into their own, connected, stable component, both to clean up the code and to remove another source of re-rendering. This will evolve into the Search Action Menu.
* Reset iOS button styles so they don't look all system-buttony.
* Fix overflowing rows pushing the X button offscreen.
* Extract bulk-locking to a shared Redux action, which reduces duplicate code.
* Replace Infusion Finder and Item Picker search bars with the new stuff.

### CODE REVIEW

File: src/app/utils/useWhatChanged.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useRef } from 'react';
+
+/**
+ * A debugging hook that will print out what changed since the last render!
+ *
+ * Example:
+ *
+ * useWhatChanged('MyComponent', {prop1, prop2, state1, state2});
+ */
+export function useWhatChanged<T extends object>(name: string, params: T) {
```
Comment (ryan-rushton) [inline src/app/utils/useWhatChanged.ts:10]: This is really cool, there has been a few times when this would have been really useful.

**Comment by ryan-rushton** [comment]
Yeah no locking updates is what I noticed too. I locked and unlocked all is:inloadout items (39) and I couldn't tell if it was actually doing anything.

**Review by ryan-rushton — APPROVED** [review]

**Comment by bhollis** [comment]
For reference I changed nothing about how bulk locking works - I just moved the code. 

The overflow is solely to handle the case in the other issue @nev-r filed where an unwrappable input (a URL) makes the delete button inaccessible.

File: src/app/utils/useWhatChanged.ts (hunk 1 above)
Comment (bhollis) [reply src/app/utils/useWhatChanged.ts:10]: Yeah I found it immediately useful! Way better than random console logs

**Pull request merged**